airdrop/airdrop.gno

```diff
@@ -185,3 +189,18 @@ func (a *Airdrop) GetConfig() Config {
 func (a *Airdrop) GetAddress() std.Address {
 	return a.address
 }
+
+// CreateSnapshot creates a new snapshot of the current state
+func (a *Airdrop) CreateSnapshot() uint64 {
+	return a.snapshotManager.CreateSnapshot(a.claimedBitmap, a.token.BalanceOf(std.GetOrigCaller()))
```

Comment:
Okay this is very confusing part.

You're currently using `std.GetOrigCaller()` to create snapshot data which means actual user have to call this function using gnokey or adena or whatever.

If this `airdrop` contract is meant to be call by directly from user this function can not be called. Of course in case of user deploy contract that uses this airdrop then it's fine scenario.

So my point is use case isn't really clear

Reply:
Then I'll change to take the caller as a parameter. something like this:

```go
// CreateSnapshot creates a new snapshot of the current state
func (a *Airdrop) CreateSnapshot(caller string) uint64 {
	callerAddr := std.Address(caller)
	return a.snapshotManager.CreateSnapshot(a.claimedBitmap, a.token.BalanceOf(callerAddr))
}
```

Reply:
https://github.com/gnoswap-labs/governance/pull/8/commits/f624c242fdb67a39c3af7220066f667575372d45

Reply:
Since above function is method not pure function, caller type can be `std.Address` to avoid extra type conversion.